### DISCUSSION TITLE
NanoPC T6 - Install the bootloader to the SPI NOR flash - Help needed.

### DISCUSSION THREAD

**Discussion by Spirarementum** [Joshua-Riek/ubuntu-rockchip#623] [Q&A]
https://github.com/Joshua-Riek/ubuntu-rockchip/wiki/NanoPC-T6
I am stuck at the very first step:

Booting directly from an NVMe requires flashing U-Boot to the SPI.
- [**OK**] Boot from an SD Card.
- **[??]**  Write the bootloader to the SPI NOR flash.
  sudo dd if=/lib/u-boot-friendlyelec-rk3588/rkspi_loader.img of=/dev/mtdblock0 conv=notrunc
  sync_

Sorry for my ignorance but where can I find the file /lib/u-boot-friendlyelec-rk3588/rkspi_loader.img ?
I have looked on the SD card on which I have written the T6 image and subsequently booted with (ubuntu-22.04.3-preinstalled-desktop-arm64-nanopc-t6.img.xz)

**Comment by wmport** [comment]
sudo dd if=/lib/u-boot/rkspi_loader.img of=/dev/mtdblock0 conv=notrunc
sync

**Comment by piit79** [comment]
You probably found it by now, but just for posterity - you can get the rkspi_loader.img file from a server linked in the [Armbian Installation Guide](https://wiki.friendlyelec.com/wiki/index.php/Armbian_Installation_Guide#Download_Required_Files) FriendlyElec Wiki page.
